packages/expo-updates/ios/EXUpdates/Events/UpdatesEventManager.swift

```diff
@@ -0,0 +1,19 @@
+//  Copyright © 2019 650 Industries. All rights reserved.
+
+import ExpoModulesCore
+
+public protocol UpdatesEventManager {
+  /**
+   The AppContext from expo-modules-core.
+   This is optional, but required for expo-updates module events to work.
+   */
+  var appContext: AppContext? { get }
+  var shouldEmitJsEvents: Bool { get }
+
+  // swift is weird and doesn't allow assigning to a property's property if anything in the setter chain is get-only (let)
+  // so we use setter methods here instead
```

Comment:
could you elaborate where's the get-only stuff in the setter chain?

Reply:
```swift
protocol IAppController {
    var eventManager: IEventManager { get }
}

class AppController: IAppController {
    let eventManager: IEventManager = EventManager()
}

protocol IEventManager {
    var appContext: String? { get set }
    
    func setAppContext(appContext: String?)
}

class EventManager: IEventManager {
    var appContext: String? = nil
    
    func setAppContext(appContext: String?) {
        self.appContext = appContext
    }
}

let testController: IAppController? = AppController()

print(testController?.eventManager.appContext ?? "")

// this works
testController?.eventManager.setAppContext(appContext: "hello")

// this errors with 
// error: cannot assign to property: 'eventManager' is a get-only property
testController?.eventManager.appContext = "hello"
```

(can input on https://swiftwasm.org/ to see output)

The compiler thinks I'm assigning to `eventManager` but in reality I'm assigning to `appContext`. No idea what's going on, but changing the set of appContext to a setter works.